File: 0965-unique-email-addresses/0965-unique-email-addresses.js
/**
 * @param {string[]} emails
 * @return {number}
 */
var numUniqueEmails = function(emails) {
    let unique=new Set()
    for (let email of emails){
        let [local,domain]=email.split("@")
        local=local.split("+")[0].replace(/\./g, "")
        unique.add(`${local}@{${domain}}`)

    }
    
    return unique.size
    
};

File: 0965-unique-email-addresses/0965-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        unique=set()

        for email in emails:
            [local,domain]=email.split("@")
            
            buildLocal=""

            for char in local:
                if char=='+':
                    break
                if char!=".":
                    buildLocal+=char
            unique.add(tuple((buildLocal,domain)))
        return len(unique)

            



    

                   
            
                





    


        
        